packages/web/pages/assets/[denom].tsx

```diff
@@ -78,28 +116,90 @@ const AssetInfoView = observer(() => {
     [assetInfoConfig]
   );
 
+  const routablePools = useRoutablePools();
+  const memoedPools = routablePools ?? [];
+
   return (
     <AssetInfoViewProvider value={contextValue}>
-      <div className="mx-auto flex max-w-container flex-col gap-8 p-8 py-4">
+      {showTradeModal && (
+        <TradeTokens
+          className="md:!p-0"
+          isOpen={showTradeModal}
+          onRequestClose={() => {
+            setShowTradeModal(false);
+          }}
+          memoedPools={routablePools ?? []}
+          swapOptions={{
+            sendTokenDenom: assetInfoConfig.denom,
+          }}
+        />
+      )}
+
+      <main className="flex flex-col gap-8 p-8 py-4">
         <Navigation />
-        <div className="flex flex-col gap-4">
-          <TokenChartSection />
+        <div className="grid grid-cols-tokenpage gap-4 xl:flex xl:flex-col">
+          <div className="flex flex-col gap-4">
+            <TokenChartSection />
+            <YourBalance denom={assetInfoConfig.denom} />
+            <TokenDetails denom={router.query.denom as string} />
+            <TwitterSection />
+          </div>
+          <div className="flex flex-col gap-4">
+            <SwapTool
+              memoedPools={memoedPools}
+              isDataLoading={!Boolean(routablePools) || isWalletLoading}
+              isInModal
+              sendTokenDenom={assetInfoConfig.denom}
+              outTokenDenom={assetInfoConfig.denom === "OSMO" ? "ATOM" : "OSMO"}
+            />
+            <RelatedAssets
+              memoedPools={memoedPools}
+              tokenDenom={assetInfoConfig.denom}
+            />
+          </div>
         </div>
-      </div>
+      </main>
     </AssetInfoViewProvider>
   );
 });
 
 const Navigation = observer(() => {
   const { assetInfoConfig } = useAssetInfoView();
+  const { chainStore } = useStore();
+  const t = useTranslation();
+  const [favoritesList, setFavoritesList] = useLocalStorageState(
```

Comment:
Would be great if we can somehow use the same favorites setting as in assets page.

Reply:
It's the same one, we use "favoritesList" key to store and retrieve data, like you do on assets page. I tried it and the states between pages seem to be synchronized, maybe I'm missing something?

Reply:
Ohh ok. Nice. Missed that.